content/docs/how_to_guides/sqlx.md

```diff
@@ -0,0 +1,71 @@
+---
+title: SQLX
+---
+
+## Overview
+
+SQLX is a powerful extension of SQL, created by Dataform. As it is an extension, every SQL file is also a valid SQLX file!
+
+## Structure
+
+SQLX contains the following components:
+
+- [Config](#Config): contains information on the purpose of the script, such as `view` or `table`.
+
+- [Pre-operations](#pre-operations): Pre operations contain SQL to be executed before the main bulk of SQL.
+
+- (Central) [SQL](sql): The main SQL operation to be performed. Later defined Javascript can be injected here, or built in functions such as [`ref()`](TODO).
+
+- [Post-operations](#post-operations): Post opertations contain SQL to be executed after the main bulk of SQL
+
+- [Javascript](#javascript-blocks): These provide all the incredible functionality of Javascript written alongside SQL!
+
+- [In-line Javascript](#in-line-javascript): In addition to Javascript blocks, in-line javascript can be written within SQL by using `${}`, for example `${console.log("foo")}`.
+
+- [Built-in functions](built-in-functions): There are various useful built-in functions that can be used, such as `ref()` or `self()`.
+
+### Config
+
+All config properties, and the config itself, are optional. TODO: Add more detail. Just link API reference?
+
+### Pre-operations
```

Comment:
Ok, I moved them under an `additional features` section and explained the caveat, just so that someone using SQLX who sees them will be able to look in the guide